src/hdmf/container.py

```diff
@@ -452,6 +452,92 @@ def __repr__(self):
                 template += "  {}: {}\n".format(k, v)
         return template
 
+    def _repr_html_(self):
+        CSS_STYLE = """
+        <style>
+            .nwb-fields { font-family: "Open Sans", Arial, sans-serif; }
+            .nwb-fields .field-value { color: #00788E; }
+            .nwb-fields details > summary { cursor: pointer; }
+            .nwb-fields details > summary:hover { color: #0A6EAA; }
+        </style>
+        """
+
+        JS_SCRIPT = """
+        <script>
+            function copyToClipboard(text) {
+                navigator.clipboard.writeText(text).then(function() {
+                    console.log('Copied to clipboard: ' + text);
+                }, function(err) {
+                    console.error('Could not copy text: ', err);
+                });
+            }
+
+            document.addEventListener('DOMContentLoaded', function() {
+                let fieldKeys = document.querySelectorAll('.nwb-fields .field-key');
+                fieldKeys.forEach(function(fieldKey) {
+                    fieldKey.addEventListener('click', function() {
+                        let accessCode = fieldKey.getAttribute('title').replace('Access code: ', '');
+                        copyToClipboard(accessCode);
+                    });
+                });
+            });
+        </script>
+        """
+        if self.__class__.__name__ == "NWBFile":
+            nwb_header_text = "NWBFile"
```

Comment:
Can you clarify why this first part of this if statement to check for "NWBFile" as a name is necessary? Isn't this covered by the second part? Also, it seems strange to check for NWBFile in HDMF.

Reply:
It's not covered by the second part because the root `NWBFile` object technically has the `name` "root".

Another option is to take this clause out let the name title of an NWBFile object be: "root (NWBFile)".

I suppose another option could be to create a method for this and override it in the `NWBFile` object.

Reply:
> let the name title of an NWBFile object be: "root (NWBFile)"

I think that would be fine. We can always customize in PyNWB if necessary. Alternatively you could also check for:

```
if self.name == 'root'`:
     nwb_header_text = self.__class__.__name__
```

but to be honest, I would go with consistency here and just let be ``root (NWBFile)``